### PR TITLE
LTD-152 Search by Product

### PR DESCRIPTION
**Pull request by saruniitr**
## Change description

Add index based on `BaseApplication` to be able search by application attributes (case_type, status etc), associated product attributes (part_number, description etc), associated parties (name, address etc).

Further required in terms of testing searches by all attributes with different keywords, combinations but this PR represents the base required to carry additional work.

### CODE REVIEW

**Comment by richtier** [comment]
can we be more explicit (but yes more verbose) with the name, go `good_on_application` instead of goa

File: api/search/goa/documents.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+@registry.register_document
+class GoodOnApplicationDocumentType(Document):
+    id = TextField()
+    quantity = FloatField()
+    value = FloatField()
+    unit = TextField()
+    item_type = TextField()
+    incorporated = BooleanField(attr="is_good_incorporated")
+    good = ObjectField(
```
Comment (richtier) [inline api/search/goa/documents.py:17]: would `InnerDoc` work here? https://elasticsearch-dsl.readthedocs.io/en/latest/persistence.html#document i.e. 

```
class Good(InnerDoc):
   id = TextField()
   ...

class GoodOnApplicationDocumentType(Document):
    ...
    good = Object(Good)
```

Comment (saruniitr) [reply api/search/goa/documents.py:17]: yes definitely, it wasn't working when I first tried so flattened it to to unblock. I will investigate further and replace it with InnerDoc

**Comment by saruniitr** [comment]
> can we be more explicit (but yes more verbose) with the name, go `good_on_application` instead of goa

Agree, couldn't sensibly shorten it so went with that.

**Review by richtier — APPROVED** [review]

File: api/search/application/documents.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    products = NestedField(
+        attr="goods",
+        properties={
+            "quantity": FloatField(),
+            "value": FloatField(),
+            "unit": TextField(),
+            "item_type": TextField(),
+            "incorporated": BooleanField(attr="is_good_incorporated"),
+            "good": ObjectField(
+                properties={
```
Comment (richtier) [inline api/search/application/documents.py:40]: we can flatten. may result in results that are easier to query and render on the frontend

```
 "description": TextField(attr="good.description"),
 "part_number": TextField(attr="good.part_number"),
...
```

File: local.env [hunk 3]
```diff
@@ -78,3 +78,5 @@ PERMISSIONS_FINDER_URL='' # required for accessing exporter lite content, can be
 LITE_API_DEMO_FLAGS_CSV="field\nteam1\nteam2"
 LITE_API_DEMO_QUEUES_CSV="field\nteam1\nteam2"
 LITE_API_DEMO_TEAMS_CSV="field\nteam1\nteam2"
+
+ELASTICSEARCH_HOST=http://localost:9200
```
Comment (r4vi) [inline local.env:82]: missing new line at end of file here

**Review by r4vi — APPROVED** [review]
LGTM :+1:

File: local.env (hunk 3 above)
Comment (saruniitr) [reply local.env]: fixed

**Pull request merged**